cxx-qt-lib: define a QHash<T> with QHashPair trait (#354)
KDAB/cxx-qt: book/src/concepts/types.md

{{#include ../../../tests/qt_types_standalone/rust/src/qset.rs:book_qset}}
```

To use key-value based container types such as `QHash<K, V>` an intermediate type on the Rust side

> is defined, this allows for implementing a trait on the key-value combination.

Reviewer: ```suggestion
is defined to implement a trait on the key-value combination.
```

Author: Done

> Then as with other container types the type name must be `QHash_K_V` as this needs

Reviewer: ```suggestion
As with other container types, the type name must be `QHash_K_V` as this needs
```

Author: Done

to match the name in the C++ code.

> So for `QHash<QString, QVariant>` an intermediate type called `QHashPair_QString_QVariant`.

Author: Done
